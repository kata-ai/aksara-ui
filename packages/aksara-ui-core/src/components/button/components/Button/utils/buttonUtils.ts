import { css } from 'styled-components';
import { colors } from '../../../../../utils/variables';
import { ButtonBaseProps } from '../types';
import { componentStyles } from '../../../../../utils';

export const ButtonDefault = css`
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonPrimary = css`
  background-color: ${colors.blue06};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.blue07};
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonSupport = css`
  border-color: ${colors.grey04};
  background-color: ${colors.grey01};
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      border-color: ${colors.grey04};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }

    &.selected {
      background-color: ${colors.blue01};
      color: ${colors.blue08};
      border-color: ${colors.blue04};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey01};
    color: ${colors.grey04};
  }
`;

export const ButtonDestructive = css`
  background-color: ${colors.red07};
  color: ${colors.grey01};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.red08};
      color: ${colors.grey01};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.red09};
      border-color: ${colors.red07};
      color: ${colors.grey01};
      box-shadow: 0 0 2px 1px rgba(186, 19, 9, 0.4), 0 0 0 1px ${colors.red07};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonGhost = css`
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonSmall = (props: ButtonBaseProps) => css`
  height: 32px;
  padding: 0 12px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 40px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 40px;'}
`;

export const ButtonMedium = (props: ButtonBaseProps) => css`
  height: 40px;
  padding: 0 16px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const ButtonLarge = (props: ButtonBaseProps) => css`
  height: 48px;
  padding: 0 24px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const InverseButton = css`
  border-color: ${colors.white};
  color: ${colors.white};

  &:disabled,
  &.disabled {
    opacity: 0.5;
  }
`;
