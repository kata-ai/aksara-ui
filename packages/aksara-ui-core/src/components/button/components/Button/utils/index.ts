export { default as renderButtonChildren } from './renderButtonChildren';
export { default as renderButtonIcon } from './renderButtonIcon';
export * from './buttonUtils';
