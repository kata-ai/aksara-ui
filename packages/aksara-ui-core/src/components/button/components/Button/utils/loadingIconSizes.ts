import { ButtonSizes } from '../types';

const loadingIconSizes = (size?: ButtonSizes) => {
  switch (size) {
    case 48:
      return 40;
    case 40:
      return 32;
    case 32:
      return 24;
    default:
      return 32;
  }
};

export default loadingIconSizes;
