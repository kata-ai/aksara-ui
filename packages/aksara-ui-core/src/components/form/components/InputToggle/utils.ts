/* eslint-disable import/prefer-default-export */
import { ToggleSizes, ToggleSizeObject } from './types';

export function toggleButtonSizes(size: ToggleSizes): ToggleSizeObject {
  if (size === 'sm') {
    return {
      width: 32,
      height: 16,
      switch: 12,
    };
  }

  return {
    width: 48,
    height: 24,
    switch: 20,
  };
}
