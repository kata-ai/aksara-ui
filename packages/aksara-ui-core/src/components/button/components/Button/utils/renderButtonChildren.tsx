import * as React from 'react';
import { colors } from '../../../../../utils';
import { InvisibleText, LoaderCircle } from '../components';
import { ButtonBaseProps, ButtonSizes } from '../types';

interface ButtonChildrenProps extends Pick<ButtonBaseProps, 'variant'> {
  isLoading?: boolean;
  size?: ButtonSizes;
  children?: React.ReactNode;
}

const loadingIconSizes = (size?: ButtonSizes) => {
  switch (size) {
    case 48:
      return 40;
    case 40:
      return 32;
    case 32:
      return 24;
    default:
      return 32;
  }
};

const renderButtonChildren = ({ isLoading, size, variant, children }: ButtonChildrenProps) => {
  if (isLoading) {
    return (
      <>
        <LoaderCircle
          size={loadingIconSizes(size)}
          buttonSize={size}
          spinnerColor={variant === 'outline' ? colors.grey04 : colors.white}
        />
        <InvisibleText>{children}</InvisibleText>
      </>
    );
  }

  return children;
};

export default renderButtonChildren;
