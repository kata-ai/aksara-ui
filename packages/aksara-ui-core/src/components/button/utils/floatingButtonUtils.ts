import { css } from 'styled-components';
import { colors } from '../../../utils';
import { ButtonSizes } from '../types';

export function floatingButtonSizes(size?: ButtonSizes) {
  switch (size) {
    case 'lg':
      return 64;
    case 'md':
      return 40;
    case 'sm':
      return 32;
    default:
      return 40;
  }
}

export const FloatingButtonSmall = css`
  height: 32px;
  width: 32px;
  font-size: 13px;
  border-radius: 32px;

  i:before {
    font-size: 13px !important;
  }

  svg {
    width: 20px !important;
    height: 20px !important;
  }
`;

export const FloatingButtonMedium = css`
  height: 40px;
  width: 40px;
  font-size: 16px;
  border-radius: 40px;

  i:before {
    font-size: 16px !important;
  }

  svg {
    width: 24px !important;
    height: 24px !important;
  }
`;

export const FloatingButtonLarge = css`
  height: 64px;
  width: 64px;
  font-size: 24px;
  border-radius: 64px;

  i:before {
    font-size: 24px !important;
  }

  svg {
    width: 32px !important;
    height: 32px !important;
  }
`;

export const FloatingButtonPrimary = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.blue08};
    }
  }
`;

export const FloatingButtonSuccess = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.green07};
    }
  }
`;

export const FloatingButtonWarning = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.yellow07};
    }
  }
`;

export const FloatingButtonDestructive = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.red07};
    }
  }
`;
