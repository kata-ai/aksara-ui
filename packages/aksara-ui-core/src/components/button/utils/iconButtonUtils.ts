import { css } from 'styled-components';
import { lighten } from 'polished';
import { themeGet } from '@styled-system/theme-get';
import { colors } from '../../../utils/variables';
import { IconButtonSizes } from '../types';

export function iconButtonSizes(size?: IconButtonSizes) {
  switch (size) {
    case 'lg':
      return 40;
    case 'md':
      return 32;
    case 'sm':
      return 24;
    default:
      return 32;
  }
}

export function iconSizes(size?: IconButtonSizes) {
  switch (size) {
    case 'lg':
      return 24;
    case 'md':
      return 16;
    case 'sm':
      return 12;
    default:
      return 20;
  }
}

export const IconButtonDefault = css`
  background-color: ${colors.gray10};
  color: ${colors.gray50};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.gray20};
      color: ${colors.gray50};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.gray70};
      color: ${colors.white};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonPrimary = css`
  background-color: ${colors.cobalt};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.semiCobalt};
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.darkCobalt};
      border-color: ${colors.darkCobalt};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonSupport = css`
  background-color: ${themeGet('colors.grey01', colors.grey01)};
  color: ${themeGet('colors.grey08', colors.grey08)};
  border-color: ${themeGet('colors.grey04', colors.grey04)};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${themeGet('colors.grey03', colors.grey03)};
      color: ${themeGet('colors.blue08', colors.blue08)};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${themeGet('colors.grey03', colors.grey03)};
      color: ${themeGet('colors.blue08', colors.blue08)};
      border-width: 2px;
      border-color: ${themeGet('colors.blue06', colors.blue06)};
      box-shadow: 0px 0px 2px rgba(0, 111, 230, 0.5);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${themeGet('colors.grey01', colors.grey01)};
    color: ${themeGet('colors.grey04', colors.grey04)};
    border-color: ${themeGet('colors.grey04', colors.grey04)};
  }
`;

export const IconButtonDestructive = css`
  background-color: ${colors.red};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: #e84f4c;
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.red};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonGhost = css`
  color: ${themeGet('colors.grey08', colors.grey08)};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover {
      color: ${themeGet('colors.blue08', colors.blue08)};
    }

    &:focus,
    &.focus {
      color: ${themeGet('colors.blue08', colors.blue08)};
      border-color: ${themeGet('colors.blue04', colors.blue04)};
      box-shadow: 0px 0px 2px rgba(0, 111, 230, 0.5);
    }

    &:active,
    &.active {
      border-color: ${themeGet('colors.indigo02', colors.indigo02)};
    }
  }

  &:disabled,
  &.disabled {
    color: ${colors.grey04};
  }
`;
