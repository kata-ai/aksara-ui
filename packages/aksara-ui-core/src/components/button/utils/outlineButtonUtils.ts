import { css } from 'styled-components';
import { colors } from '../../../utils/variables';
import { componentStyles } from '../../../utils';
import { OutlineButtonBaseProps } from '../types';

export const OutlineButtonDefault = css`
  color: ${colors.grey08};
`;

export const OutlineButtonPrimary = css`
  color: ${colors.blue07};
`;

export const OutlineButtonSuccess = css`
  color: ${colors.green07};
`;

export const OutlineButtonWarning = css`
  color: ${colors.yellow07};
`;

export const OutlineButtonDestructive = css`
  color: ${colors.red07};
`;

export const OutlineButtonSmall = (props: OutlineButtonBaseProps) => css`
  height: 32px;
  padding: 0 12px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 40px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 40px;'}
`;

export const OutlineButtonMedium = (props: OutlineButtonBaseProps) => css`
  height: 40px;
  padding: 0 24px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const OutlineButtonLarge = (props: OutlineButtonBaseProps) => css`
  height: 56px;
  padding: 0 24px;
  font-size: ${componentStyles.text[400].fontSize};
  line-height: ${componentStyles.text[400].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const InverseButton = css`
  border-color: ${colors.grey01};
  color: ${colors.grey01};

  &:disabled,
  &.disabled {
    opacity: 0.5;
  }
`;
