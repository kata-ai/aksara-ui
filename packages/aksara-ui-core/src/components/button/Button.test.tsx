import * as React from 'react';
import { render } from '@testing-library/react';
import { IconDocAdd } from '@aksara-ui/icons';

import { colors } from '../../utils';
import { Button, FloatingButton, IconButton, OutlineButton } from './index';

describe('components/Button', () => {
  describe('<Button />', () => {
    test('renders correctly', () => {
      const { container } = render(<Button>example</Button>);

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders colors correctly by variant', () => {
      const { container } = render(<Button variant="primary">test button</Button>);

      expect(container.firstChild).toHaveStyleRule('background-color', colors.cobalt);
      expect(container.firstChild).toHaveStyleRule('color', colors.white);
    });

    test('renders block buttons correctly', () => {
      const { container } = render(<Button block>test button</Button>);

      expect(container.firstChild).toHaveStyleRule('display', 'block');
      expect(container.firstChild).toHaveStyleRule('width', '100%');
    });

    test('renders icons correctly', () => {
      const { container } = render(<Button icon="add">test button</Button>);
      const icon = container.querySelector('i');

      expect(icon).toBeInTheDocument();
    });

    test('renders icons correctly with ReactNode', () => {
      const { container } = render(
        <Button size="sm" icon={IconDocAdd} iconPosition="right">
          test button
        </Button>
      );
      const icon = container.querySelector('svg');

      expect(icon).toBeInTheDocument();
    });
  });

  describe('<OutlineButton />', () => {
    test('renders correctly', () => {
      const { container } = render(<OutlineButton>example</OutlineButton>);

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders block buttons correctly', () => {
      const { container } = render(<OutlineButton block>test button</OutlineButton>);

      expect(container.firstChild).toHaveStyleRule('display', 'block');
      expect(container.firstChild).toHaveStyleRule('width', '100%');
    });

    test('renders icons correctly', () => {
      const { container } = render(<OutlineButton icon="add">test button</OutlineButton>);
      const icon = container.querySelector('i');

      expect(icon).toBeInTheDocument();
    });
  });

  describe('<FloatingButton />', () => {
    test('renders correctly', () => {
      const { container } = render(
        <FloatingButton>
          <i className="icon-trash" />
        </FloatingButton>
      );

      expect(container.firstChild).toBeInTheDocument();
    });
  });

  describe('<IconButton />', () => {
    test('renders correctly', () => {
      const { container } = render(
        <IconButton>
          <i className="icon-trash" />
        </IconButton>
      );

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders in different sizes', () => {
      const { container } = render(
        <>
          <IconButton size="sm">
            <i className="icon-trash" />
          </IconButton>
          <IconButton size="md">
            <i className="icon-trash" />
          </IconButton>
        </>
      );

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders colors correctly by variant', () => {
      const { container } = render(
        <IconButton variant="primary">
          <i className="icon-add" />
        </IconButton>
      );

      expect(container.firstChild).toHaveStyleRule('background-color', colors.cobalt);
      expect(container.firstChild).toHaveStyleRule('color', colors.white);
    });
  });
});
