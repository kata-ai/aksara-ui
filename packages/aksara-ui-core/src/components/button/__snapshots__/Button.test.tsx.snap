// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button <Button /> renders block buttons correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  line-height: 1;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  border: 1px solid transparent;
  display: block;
  width: 100%;
  position: relative;
  color: #313f4e;
  height: 40px;
  padding: 0 16px;
  font-size: 14px;
  line-height: 20px;
  border-radius: 4px;
}

.c0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 0px;
}

.c0:not(:disabled):not(.disabled):hover,
.c0:not(:disabled):not(.disabled).hover,
.c0:not(:disabled):not(.disabled):focus,
.c0:not(:disabled):not(.disabled).focus {
  background-color: #ebedef;
  color: #004793;
}

.c0:not(:disabled):not(.disabled):focus,
.c0:not(:disabled):not(.disabled).focus,
.c0:not(:disabled):not(.disabled):active,
.c0:not(:disabled):not(.disabled).active {
  background-color: #ebedef;
  color: #004793;
  border-color: #006fe6;
  box-shadow: 0 0 2px 1px rgba(0,111,230,0.4),0 0 0 1px #006fe6;
}

.c0:disabled,
.c0.disabled {
  border-color: #d7dbdf;
  background-color: #d7dbdf;
  color: #ffffff;
}

<button
  class="c0"
  display="block"
  width="100%"
>
  test button
</button>
`;

exports[`components/Button <Button /> renders colors correctly by variant 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  line-height: 1;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  border: 1px solid transparent;
  display: inline-block;
  position: relative;
  background-color: #006fe6;
  color: #fff;
  height: 40px;
  padding: 0 16px;
  font-size: 14px;
  line-height: 20px;
  border-radius: 4px;
}

.c0:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 0px;
}

.c0:not(:disabled):not(.disabled):hover,
.c0:not(:disabled):not(.disabled).hover,
.c0:not(:disabled):not(.disabled):focus,
.c0:not(:disabled):not(.disabled).focus {
  background-color: #005bbd;
  color: #fff;
}

.c0:not(:disabled):not(.disabled):focus,
.c0:not(:disabled):not(.disabled).focus,
.c0:not(:disabled):not(.disabled):active,
.c0:not(:disabled):not(.disabled).active {
  background-color: #004793;
  border-color: #006fe6;
  box-shadow: 0 0 2px 1px rgba(0,111,230,0.4),0 0 0 1px #006fe6;
}

.c0:disabled,
.c0.disabled {
  border-color: #d7dbdf;
  background-color: #d7dbdf;
  color: #ffffff;
}

<button
  class="c0"
  display="inline-block"
>
  test button
</button>
`;
